src/Components/BehaviorsModal/Internal/BehaviorSection/BehaviorVisualRuleSection.tsx

```diff
@@ -0,0 +1,175 @@
+import { BehaviorsModalContext } from '../../BehaviorsModal';
+import WidgetsContainer from '../Widgets/WidgetsContainer';
+import { getStyles } from './BehaviorSection.styles';
+import { useId } from '@fluentui/react-hooks';
+import ColorPillsTooltip from '../../../ColorPillsTooltip/ColorPillsTooltip';
+import { useExtendedTheme } from '../../../../Models/Hooks/useExtendedTheme';
+
+export interface IBehaviorsSectionProps {
+    behavior: IBehavior;
+}
+
+export interface VisualColorings {
+    color: string;
+    label?: string;
+}
+
+export interface VisualBadges {
+    color: string;
+    iconName: string;
+    labelExpression?: string;
+}
+
+const BehaviorVisualRuleSection: React.FC<IBehaviorsSectionProps> = ({
+    behavior
+}) => {
+    const styles = getStyles();
+    const { twins, mode } = useContext(BehaviorsModalContext);
+    const badgeIdPrefix = useId('cb-visual-rule-badge');
+    const tooltipId = useId('cb-behavior-section-header-tooltip');
+    const theme = useExtendedTheme();
+
+    const { meshColorings, badges } = useMemo(() => {
+        const rules =
+            behavior.visuals.filter(ViewerConfigUtility.isVisualRule) || [];
+        const visualColorings: VisualColorings[] = [];
+        const visualBadges: VisualBadges[] = [];
+        rules.forEach((rule) => {
+            rule.valueRanges.forEach((condition) => {
+                if (
+                    shouldShowVisual(
+                        rule.valueRangeType,
+                        twins,
+                        rule.valueExpression,
+                        condition.values as any
+                    )
+                ) {
+                    if (hasBadge(condition.visual.iconName)) {
+                        visualBadges.push({
+                            color: condition.visual.color,
+                            iconName: condition.visual.iconName,
+                            labelExpression: condition.visual.labelExpression
+                        });
+                    } else {
+                        visualColorings.push({
+                            color: condition.visual.color,
+                            label: condition.visual.labelExpression
+                        });
+                    }
+                }
+            });
+        });
+        return {
+            meshColorings: visualColorings,
+            badges: visualBadges
+        };
+    }, [behavior.visuals, mode, twins]);
+
+    const popoverVisual = useMemo(
+        () => behavior.visuals.filter(ViewerConfigUtility.isPopoverVisual)[0],
+        [behavior]
+    );
+
+    const tooltipContent = (colorings: VisualColorings[]) => {
+        return <ColorPillsTooltip visualColorings={colorings} />;
+    };
+
+    return (
+        <>
+            <div className={styles.behaviorSection}>
+                <TooltipHost
+                    id={tooltipId}
+                    tooltipProps={{
+                        onRenderContent: () => tooltipContent(meshColorings)
```

Comment:
This is a different way of creating a tooltip. TooltipHost wraps around the target element and brings onHover/onFocus capabilities with it. onHover of the element being wraped around it will show the tooltip, to render custom content you need to pass the onRenderFunction.